parcels/fieldfilebuffer.py

```diff
@@ -574,8 +574,6 @@ def data_access(self):
                 else:
                     if not self.autochunkingfailed:
                         data = data.rechunk(self.chunk_mapping)
-                    if self.rechunk_callback_fields is not None:
-                        self.rechunk_callback_fields()
```

Comment:
actually, if we now do have the failsafe active, then this callback needs to stay (in my opinion). Reason:

The failsafe comes into play if the user requests `chunksize='auto'`, does not provide a dimension mapping netcdf <-> parcels, and the dimension mapping is not in the static maps (i.e. the dimension name mappings that we know). In that case, we either fully abort, or we do the failsafe (i.e. parse the netcdf file and match the dimensions heuristically). In that case, the `self.chunksize` will be overwritten, from `'auto'` to a safe, specific, strictly 2D-spatial chunksize. Thus, the check in l. 558 will return `False` even though `'auto'` was requested by the user. Now, the field attached to the file doesn't know at this point that we overruled the auto-decision - its chunk information are still uninitialised. Hence, they need to be updated by the callback in either case. And that was the purpose of this line.

I hope you can follow the trail of thoughts - decide for yourself. In my opinion, that line needs to stay, but there's a counter-argument to it, I'm happy to hear it.

Reply:
This is an outdated change, I reverted it already in a later commit (593cf10). The discussion about how to call this call_back_fields is better contained in #976